#!/usr/bin/env python
PACKAGE = "mambo_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("telemetry_rate_hz", double_t, 0,
    "Frequency of pulling+publishing state updates from drone (1/sec)",
    5.0, 0.1, 5.0)
gen.add("cmd_timeout_sec", double_t, 0,
    "Timeout for repeatedly sending takeoff/land/estop commands (sec)",
    5.0, 0.01, 10.0)
gen.add("sleep_quanta_sec", double_t, 0,
    "Internal operations - base unit to sleep in-between processing BLE commands [AFFECTS TELEMETRY_RATE_HZ] (sec)",
    0.1, 0.001, 0.2)
gen.add("max_vert_speed_mps", double_t, 0,
    "Constraint on maximum vertical speed (m/s)",
    1.0, 0.0, 2.0)
gen.add("max_tilt_deg", double_t, 0,
    "Constraint on maximum lateral tilt (deg)",
    45.0, 0.0, 90.0)

exit(gen.generate(PACKAGE, "mambo_driver_node.py", "Mambo"))